microcosm_pubsub/handlers/chain_handlers.py

```diff
@@ -0,0 +1,49 @@
+from microcosm_pubsub.chain import Chain
+from microcosm_pubsub.handlers.uri_handler import URIHandler
+from abc import ABCMeta, abstractmethod
+
+
+class ChainHandler(metaclass=ABCMeta):
+    """
+    Resolve a chain on call. Pass to the chain the message to the chain.
+
+    """
+    @abstractmethod
+    def get_chain(self):
+        pass
+
+    def __call__(self, message):
+        return Chain(
+            self.get_chain(),
+        ).resolve(
+            message=message,
+        )
+
+
+class ChainURIHandler(URIHandler, metaclass=ABCMeta):
+    """
+    Base handler for URI-driven events based on URIHandler
+    Resolve a chain on handle.
+    Pass to the chain the message and the fetched resource.
+
+    """
+
+    @abstractmethod
+    def get_chain(self):
+        pass
+
+    @property
+    def resource_name(self):
+        return "resource"
+
+    def handle(self, message, uri, resource):
+        kwargs = dict(message=message)
+        kwargs[self.resource_name] = resource
+
+        res = Chain(
+            self.get_chain(),
+        ).resolve(
+            **kwargs,
+        )
+        if res is None:
```

Comment:
Most chains don't really return any value, but `URIHandler` expects something. Instead of adding `return True` step to every handler - i've added this

Reply:
Maybe we shouldn't rely on the return value and always return `True`?

The chain can raise `SkipMessage` right?

Reply:
OK. We will lose some of the URIHandler functionality - but i'm fine with it.
Yes.